### DISCUSSION TITLE
Webkit not found

### DISCUSSION THREAD

**Discussion by Benjamin-Re** [cypress-io/cypress#29912] [Docker] [answered]
Hello,

I can't run the tests in webkit in a docker container running Debian 12.

When I try `npm cypress run --browser=/<path-to-webkit-binary>

 i get the error `Unable to find browser with path /<path-to-webkit-binary>
We could not identify a known browser at the path you provided: <path-to-webkit-binary>.` But the webkit binary is in this location and that is the same way that we launch Cypress with Chromium by pointing --browser to the binary. 

I added the `experimentalWebKitSupport: true ` line to the Cypress config.

In the Docker file I am getting playwright webkit 1.37. In the Cypress docs it says to download 1.34 but that playwright version doesn't support Debian 12.

In the Docker image I'm installing a missing dependency that the binary expects but isn't downloaded already. Then I'm creating a symlink for a library that it expects but which exact version I can't get. And I add the paths where the playwright webkit installation includes libraries to the environment variable LD_LIBRARY_PATH.
 
```
apt-get update && apt-get install libwebkit2gtk-4.1-0 -y
RUN PLAYWRIGHT_DOWNLOAD_HOST=https://<someurl>/repository/raw-proxy-playwright \
    PLAYWRIGHT_BROWSERS_PATH=/opt/ \
    npx playwright install --with-deps webkit \
  && ln -s /opt/webkit-2035/minibrowser-gtk/bin/MiniBrowser /usr/local/bin/webkit 

ln -s /usr/lib/x86_64-linux-gnu/libjxl.so.0.7 /usr/lib/x86_64-linux-gnu/libjxl.so.0.8

ENV LD_LIBRARY_PATH="/usr/local/bin/webkit-2035/minibrowser-gtk/lib:$LD_LIBRARY_PATH"
ENV LD_LIBRARY_PATH="/usr/local/bin/webkit-2035/minibrowser-gtk/sys/lib/:$LD_LIBRARY_PATH"
```

`ldd webkit` shows that all dependencies are found

`webkit -version` shows 
`(webkit:4022): Gtk-WARNING **: 13:58:39.779: cannot open display` 
But maybe this is not an issue since it should run in headless mode. 

It works on Mac with the open option. And I installed playwright on the container and ran an e2e test using webkit with it. So the binary should be ok. 

I also tried different other commands to launch Cypress with webkit:
`CYPRESS_BROWSER_PATH=/<path-to-webkit-binary> npm run cypress' -> this launches electron
`CYPRESS_BROWSER_PATH=/<path-to-webkit-binary> npm run cypress --browser=webkit' -> `Browser: webkit was not found on your system or is not supported by Cypress.`

I added the binary path to the PATH variable export PATH=$PATH:<path-to-webkit>
then `npm run cypress --browser=webkit` -->  `Browser: webkit was not found on your system or is not supported by Cypress.`

I also tried to add webkit to the browsers array in cypress.config.ts
                const webkit: Cypress.Browser =     {
                    name: 'webkit',
                    channel: 'stable',
                    family: 'webkit',
                    displayName: 'WebKit',
                    version: '17',
                    path: '/usr/local/bin/webkit',
                    majorVersion: '17',
                    warning: 'WebKit support is currently experimental. Some functions may not work as expected.',
                    isHeadless: true,
                    isHeaded: false
                  }
                config.browsers.push(webkit)
But the error is the same
`Browser: webkit was not found on your system or is not supported by Cypress.`

If you have any idea what I could try, please let me know.

**Comment by Benjamin-Re** [comment]
The fix was to add playwright-webkit to package.json as dependency but then tell yarn that we already have the binary through an environment variable to not download it again from the internet.

**Comment by MikeMcC399** [comment, marked as answer]
This question was also asked in Discord. I posted the following steps to demonstrate usage in a Cypress Docker container:

```shell
git clone https://github.com/MikeMcC399/github-action --branch webkit
cd github-action/examples/webkit
npm ci
docker run -it --rm -v .:/e2e -w /e2e cypress/base
```
```shell
unset CI
npx playwright install-deps webkit
npx playwright install
npx cypress install
npx cypress run --browser webkit
```

**Comment by MikeMcC399** [comment]
@Benjamin-Re 

Thanks for your comment about Debian 12 compatibility!

- I submitted https://github.com/cypress-io/cypress-documentation/pull/5884 to suggest removing `1.34` from the installation instructions.

**Comment by MikeMcC399** [comment]
- I also added https://github.com/cypress-io/cypress-docker-images/issues/1183
